[Siddharth Srivastava] iP

Siddharth-Sid opened the pull request:
### DukePro :sunglasses: 
> Creativity is Intelligence having fun - Einstein ([Source](https://quoteinvestigator.com/2017/03/02/fun/))

DukePro frees your mind of having to remember things you need to do. It's,

- Text based
- VERY HARD to use
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. Download it from [here](https://github.com/Siddharth-Sid/ip)
2. Run Gradle to build the project
3. Run Duke.main()

#### Features:

- [x] Adding Tasks
- [x] Adding Deadlines
- [x] Adding Events
- [ ] Managing Deadline (Coming Soon)
- [ ] Reminders (Coming Soon)

If you Java programmer, you can use it to practice Java too. Here's the ```main``` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

sidcodegen committed:
Level 1. Greet, Echo, Exit

sidcodegen committed:
Level 2. Add, List

sidcodegen committed:
Level 3. Mark as Done

sidcodegen committed:
Level 4. ToDos, Events, Deadlines

sidcodegen committed:
A-TextUiTesting: Automated Text UI Testing

sidcodegen committed:
Level 5. Handle Errors

sidcodegen committed:
Level 6. Delete

sidcodegen committed:
A-Enums: Use Enums

sidcodegen committed:
Level 7. Save

sidcodegen committed:
Level 8. Dates and Times

sidcodegen committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

sidcodegen committed:
A-MoreOOP: Use More OOP

sidcodegen committed:
A-Packages: Organize into Packages

sidcodegen committed:
A-JUnit: Add JUnit Tests

sidcodegen committed:
A-JavaDoc: JavaDoc

sidcodegen committed:
A-CodingStandard: Follow the Coding Standard

sidcodegen committed:
Level-9: Find

sidcodegen committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java

sidcodegen committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Command.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java

TaoJun99 commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,295 @@
+     * @param tasks The Object to contain the List of the Tasks
+     * @throws DukeException An Exception class to be thrown if the command taskDetails is not valid
+     */
+    private void displayTaskList(String taskDetails, UI ui, TaskList tasks) throws DukeException {
```
taskDetails is not a collection of items so maybe it can be renamed to taskDetail instead?

Siddharth-Sid replied:
Absolutely

TaoJun99 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,168 @@
+            }
+        } catch (IOException e) {
+            File deletedFile = new File("data/duke.txt");
+            boolean isDeleted = deletedFile.delete(), shouldShow = false;
```
Maybe canShow is more appropriate than shouldShow for a boolean?

Siddharth-Sid replied:
Both Seem similar so will discuss with tutors and accordingly change. Thank you for pointing it out.

TaoJun99 commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,295 @@
+     * @param tasks The Object to contain the List of the Tasks
+     * @throws DukeException An Exception class to be thrown if the command taskDetails is not valid
+     */
+    private void addTodo(String taskDetails, UI ui, TaskList tasks) throws DukeException {
```
Same as the previous comment, taskDetail instead of taskDetails?

Siddharth-Sid replied:
Understood.

TaoJun99 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,86 @@
+ */
+public class TaskList {
+
+    private final List<Task> taskList;
```
Maybe tasks is more appropriate than taskList since it is a collection?

Siddharth-Sid replied:
Sure

TaoJun99 reviewed (commented):
Overall, there are very few coding standard violations except for some minor ones 👍

sidcodegen committed:
Merge commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d' into gradle

sidcodegen committed:
A-CheckStyle: Use CheckStyle

luodan01 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,51 @@
+    /**
+     * Method to return the specific Command processed from the given from the Command String
+     *
+     * @param taskDescription The Command inputted as a String
```
Minor error, 'inputted' could maybe be changed to 'input'. In addition, perhaps a better variable name would be something like ```userInput``` as I feel that ```taskDescription``` is not the most intuitive, and could be construed as the task details.

luodan01 commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,298 @@
+    /**
+     * Enum to encapsulate the type of Commands
+     */
+    public enum Commands {
```
Perhaps this could be renamed to ```CommandType```? Plural form doesn't seem very intuitive

luodan01 commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,298 @@
+     * @param ui The Object to render the UI
+     * @throws DukeException An Exception class to be thrown if the command taskDetails is not valid
+     */
+    private void goodbye(String taskDetails, UI ui) throws DukeException {
```
Perhaps this could be renamed to sound like an action?

luodan01 commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,298 @@
+    /**
+     * Constructor for the Class Command
+     *
+     * @param type The type of Command
+     * @param taskDetails The String describing the various attributes for the command
```
Perhaps it is better to add a full stop behind each parameter description?

Siddharth-Sid replied:
Ah ok

luodan01 commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,298 @@
+        case DEADLINE:
+            addDeadline(taskDetails, ui, tasks);
+            break;
+
```
Perhaps it's good to remove this line break?

Siddharth-Sid replied:
Sure

luodan01 commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,298 @@
+            goodbye(taskDetails, ui);
+            storage.saveTaskList(tasks);
+            break;
+        case DEADLINE:
```
Perhaps it would be good to name this ```ADD_DEADLINE```? I feel that ```DEADLINE``` is a bit confusing (same for ```EVENT``` and ```TODO```) since it doesn't sound like a command

luodan01 commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,88 @@
+/**
+ * Class to abstract the rendering of the UI of the Duke
+ */
+public class UI {
```
Perhaps it would be better to use pascal case and name the class ```Ui```?

luodan01 commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,88 @@
+    public static final String ANSI_RESET = "\u001B[0m";
+    public static final String ANSI_CYAN = "\u001B[36m";
+    public static final String ANSI_RED = "\u001B[31m";
+    private final Scanner input;
```
Perhaps this could be renamed to something like ```scanner``` or ```sc```?

luodan01 commented on src/main/java/duke/task/EventTask.java:
```diff
@@ -0,0 +1,32 @@
+
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (at: " + details.format(DateTimeFormatter.ofPattern("MMM dd yyyy, HH:mm"))
+                + ")";
```
Perhaps it would be better if ```String.format``` was used here?

luodan01 reviewed (commented):
LGTM! I like your comprehensive tests and how your code is organised. 🙂

sidcodegen committed:
Level 10. GUI

sidcodegen committed:
Improving the GUI

sidcodegen committed:
Add assertion checks for TaskList

Currently the Task List does not check the index for getTask,
completeTask and deleteTask functions.

Currently the Storage class does not have a check for the TastList being
uninitialised in the Load function

Let's add assertion checks verify if the index is within the limit
(0 - Size) in the Task List class and also place a null check for the
Task List in the Storage class for the Storage.load function.

sidcodegen committed:
Make changes in FXML files to improve GUI

Currently the messages have a fixed height and there is no colour
difference for the user or duke messages

Let us add a colour parameter in the dialog boxes and update the FXML
file to add an alignment field and remove the max Height and max Width
fields

sidcodegen committed:
Modified Test Cases
Currently the UI for Find command has no mention of the search parameter
so updated the UI to show new string and accordingly updated the
expected string in the test cases

sidcodegen committed:
Improve Code Quality

Currently there are a couple of Javadoc's missing and some code
fragments need to be refactored to improve code quality by following
SLAP, removing arrowheads, avoiding empty catch statements, renaming
variables and add parenthesis

Let's
1) Add Javadoc's to DeadlineTask, EventTask and Task class constructors
2) Add Comments in empty Catch Blocks in Storage class
3) Rename variables in Dialog Box Class
4) Remove arrowhead from MainWindow and Storage Class
5) Add parenthesis in the Parser Class

Siddharth-Sid committed:
Merge pull request #2 from Siddharth-Sid/branch-A-CodeQuality

Improve Code Quality

sidcodegen committed:
Add Extension B-DoWithinPeriodTasks

Siddharth-Sid committed:
Merge pull request #3 from Siddharth-Sid/heads/BCD-Extension

Add Extension B-DoWithinPeriodTasks

sidcodegen committed:
Small Changes to GUI

Siddharth-Sid committed:
Update README.md

sidcodegen committed:
User Guide

Siddharth-Sid committed:
Update README.md

sidcodegen committed:
Merge branch 'master' of https://github.com/Siddharth-Sid/ip

Siddharth-Sid committed:
Update README.md

sidcodegen committed:
UI image added

sidcodegen committed:
Merge branch 'master' of https://github.com/Siddharth-Sid/ip

sidcodegen committed:
Renamed UI.png to Ui.png

sidcodegen committed:
changed file location

sidcodegen committed:
Small code quality issues fixed

Siddharth-Sid committed:
Set theme jekyll-theme-slate
